src/test/Extends.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { Equals } from "../Equals";
+import type { Extends } from "../Extends";
+import { assert } from "../assert";
+
+const emptyObj = {};
+
+type EmptyObj = typeof emptyObj;
+
+type Fn<A extends any[] = any[], R = any> = (...args: A) => R;
+
+type Obj = Record<string, unknown>;
+
+function fn() {
+    return undefined;
+}
+
+class A {
+    a = 1;
+}
+
+{
+    // identity
+    assert<Equals<Extends<any, any>, true>>();
+    assert<Equals<Extends<never, never>, true>>();
+    assert<Equals<Extends<unknown, unknown>, true>>();
+    assert<Equals<Extends<void, void>, true>>();
+    assert<Equals<Extends<undefined, undefined>, true>>();
+    assert<Equals<Extends<null, null>, true>>();
+    assert<Equals<Extends<boolean, boolean>, true>>();
+    assert<Equals<Extends<number, number>, true>>();
+    assert<Equals<Extends<string, string>, true>>();
+    assert<Equals<Extends<symbol, symbol>, true>>();
+    assert<Equals<Extends<any[], any[]>, true>>();
+    assert<Equals<Extends<Fn, Fn>, true>>();
+    assert<Equals<Extends<Obj, Obj>, true>>();
+    assert<Equals<Extends<A, A>, true>>();
+}
+
+{
```

Comment:
Just a little remark regarding the following tests - I intentionally added duplications because it is important in the respective context to understand the whole picture. Overlapping tests do not really harm.

Reply:
Got you!